### ISSUE TITLE
Prevent noisy sentry alerts

### ISSUE THREAD

**Issue by Datamance** [lookit/lookit-api#483]
**TL;DR**: Correctly order the updating of StatefulSets and Deployments in lookit-orchestrator to avoid transient boot errors.

**Narrative**
As a developer, I want sentry alerts to be as meaningful as possible. Right now we have transient deployment errors that are an artifact of certain pieces going live and expecting other systems to be up before them. Notable examples include:

[**TLSParameterError in celery workers** ](https://sentry.io/share/issue/2e9a2ae537424baab52c1daef7cc7f92/) Docker container with actual certs hasn't created them and put them in shared volume yet

[**Celery consumer can't connect to AMPQ broker**](https://sentry.io/share/issue/44f6dc909d50497399cdef78044ce466/)
rabbitmq statefulset hasn't started up before Celery worker

[**Psycopg2 can't connect to database**](https://sentry.io/share/issue/1ee2102c82be4abbb7b9496228a12b3a/)
cloudsql proxy pod hasn't started up before anything that uses it

**Acceptance Criteria**

- [x] The three issues above stop happening on redeploys

Once these issues are resolved, we can do the final step in #466 .

**Comment by Datamance** [comment]
Attempt #1 to mitigate this issue was using the `initContainers` strategy detailed [here](https://kubernetes.io/docs/concepts/workloads/pods/init-containers/#init-containers-in-use). This doesn't seem to do the job quite well enough, so I'm going to reach out for help in Slack and see if I can get anyone to bite.

**Comment by Datamance** [comment]
Carryover into #486

**Comment by Datamance** [comment]
Nevermind, errors are still being thrown and I understand why now -

`until nslookup $(RABBITMQ_SERVICE_NAME).$(cat /var/run/secrets/kubernetes.io/serviceaccount/namespace).svc.cluster.local; do echo waiting for $(RABBITMQ_SERVICE_NAME); sleep 2; done`

Is going to check the DNS for the service, which would work... if the service was never up in the first place. Instead, the initContainer `nslookup` command is seeing the DNS record from the pre-deploy service. The command is basically useless as such - we need to actually order the deployments and do true remote healthchecks.

I have to rethink how to accomplish this.

**Comment by Datamance** [comment]
Solved by https://github.com/lookit/lookit-orchestrator/commit/111e2a0bb7e5437c4a0860d3eef0080d304f4510

Instead of checking DNS, we have the initContainers run `kubectl rollout status`, using proper RBAC with a deployment/statefulset "status reader" service account patched into the dependent deployments and statefulsets.
